plugins/parsers/json/parser_test.go

```diff
@@ -440,3 +441,34 @@ func TestHttpJsonBOM(t *testing.T) {
 	_, err := parser.Parse(jsonBOM)
 	assert.NoError(t, err)
 }
+
+//for testing issue #4260
+func TestJSONParseNestedArray(t *testing.T) {
+	testString := `{
+	"total_devices": 5,
+	"total_threads": 10,
+	"shares": {
+	"total": 5,
+	"accepted": 5,
+	"rejected": 0,
+	"avg_find_time": 4,
+	"tester": "work",
+	"tester2": "don't want this",
+	"tester3": 7.93
+	}
+	}`
+
+	parser := JSONParser{
+		MetricName: "json_test",
+		TagKeys:    []string{"total_devices", "total_threads", "shares_tester", "shares_tester3"},
+		//TagKeys: []string{"mytag", "a", "b_c", "b_d"},
+	}
+
+	metrics, err := parser.Parse([]byte(testString))
+	t.Logf("error: %v", err)
+	t.Logf("num of metrics: %v", len(metrics))
+	t.Logf("metric tags: %v", metrics[0].Tags())
+	t.Logf("metric fields: %v", metrics[0].Fields())
```

Comment:
Instead of logging the values, use require to assert that the values are as expected.